docs/source/user_guide/installation/install-from-docker.rst

```diff
@@ -0,0 +1,199 @@
+
+            docker run -it pybamm/pybamm:jax
+
+.. tab:: IDAKLU Solver
+
+      .. code:: bash
+
+            docker run -it pybamm/pybamm:idaklu
+
+.. tab:: All Solver
+
+      .. code:: bash
+
+            docker run -it pybamm/pybamm:all
+
+2. You will now be inside the Docker container's shell. You can use PyBaMM and its dependencies as if you were in a virtual environment.
+
+3. You can execute PyBaMM-related commands, run tests develop & contribute from the container.
+
+Exiting the Docker Container
+----------------------------
+
+To exit the Docker container's shell, you can simply type:
+
+.. code-block:: bash
+
+      exit
+
+This will return you to your host machine's terminal.
+
+Building Docker Image Locally from Source
+-----------------------------------------
+
+If you want to build the PyBaMM Docker image locally from the PyBaMM source code, follow these steps:
+
+1. Clone the PyBaMM GitHub repository to your local machine if you haven't already:
+
+.. code-block:: bash
+
+      git clone https://github.com/pybamm-team/PyBaMM.git
+
+2. Change into the PyBaMM directory:
+
+.. code-block:: bash
+
+      cd PyBaMM
+
+3. Build the Docker image using the following command:
+
+.. code-block:: bash
+
+      docker build -t pybamm -f scripts/Dockerfile .
+
+4. Once the image is built, you can run a Docker container using:
+
+.. code-block:: bash
+
+      docker run -it pybamm
+
+Building Docker Images with Optional Args
+-----------------------------------------
+
+When building the PyBaMM Docker images locally, you have the option to include specific solvers by using optional arguments. These solvers include:
+
+- IDAKLU: For IDA solver provided by the SUNDIALS plus KLU.
+- ODES: For scikits.odes solver for ODE & DAE problems.
+- JAX: For Jax solver.
+
+To build the Docker images with optional arguments, you can follow these steps for each solver:
+
+Build Docker Image with IDAKLU Solver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Follow the same steps as above to clone the PyBaMM repository and navigate to the source code directory.
+
+3. Build the Docker image for IDAKLU using the following command:
+
+.. code-block:: bash
+
+      docker build -t pybamm:idaklu -f scripts/Dockerfile --build-arg IDAKLU=true .
+
+Build Docker Image with ODES Solver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Follow the same steps as above to clone the PyBaMM repository and navigate to the source code directory.
+
+2. Build the Docker image for ODES using the following command:
+
+.. code-block:: bash
+
+      docker build -t pybamm:odes -f scripts/Dockerfile --build-arg ODES=true .
+
+Build Docker Image with JAX Solver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Follow the same steps as above to clone the PyBaMM repository and navigate to the source code directory.
+
+2. Build the Docker image for JAX using the following command:
+
+.. code-block:: bash
+
+      docker build -t pybamm:jax -f scripts/Dockerfile --build-arg JAX=true .
```

Comment:
Could you also add the `ALL` arg here? I think it would make sense to add a new arg to the Dockerfile itself called `ALL` that installs every solver-specific dependency!

Reply:
> Could you also add the `ALL` arg here? I think it would make sense to add a new arg to the Dockerfile itself called `ALL` that installs every solver-specific dependency!

Sounds good to me, will do it.